azure-mgmt-graph-rbac/src/main/java/com/microsoft/azure/management/graphrbac/Application.java

```diff
@@ -0,0 +1,294 @@
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasName;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import com.microsoft.azure.management.resources.fluentcore.model.Indexable;
+import com.microsoft.azure.management.resources.fluentcore.model.Updatable;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An immutable client-side representation of an Azure AD application.
+ */
+@Fluent(ContainerName = "/Microsoft.Azure.Management.Fluent.Graph.RBAC")
+@Beta
+public interface Application extends
+        Indexable,
+        HasInner<ApplicationInner>,
+        HasId,
+        HasName,
+        Updatable<Application.Update> {
+    /**
+     * @return the object type
+     */
+    String objectType();
+
+    /**
+     * @return the application ID
+     */
+    String appId();
+
+    /**
+     * @return the application permissions
+     */
+    List<String> appPermissions();
+
+    /**
+     * @return whether the application is be available to other tenants
+     */
+    boolean availableToOtherTenants();
+
+    /**
+     * @return a collection of URIs for the application
+     */
+    List<String> identifierUris();
+
+    /**
+     * @return a collection of reply URLs for the application
+     */
+    List<String> replyUrls();
+
+    /**
+     * @return the home page of the application
+     */
+    String homepage();
+
+    /**
+     * @return the mapping of password credentials from their names
+     */
+    Map<String, PasswordCredential> passwordCredentials();
+
+    /**
+     * @return the mapping of certificate credentials from their names
+     */
+    Map<String, CertificateCredential> certificateCredentials();
+
+    /**************************************************************
+     * Fluent interfaces to provision an application
+     **************************************************************/
+
+    /**
+     * Container interface for all the definitions that need to be implemented.
+     */
+    interface Definition extends
+            DefinitionStages.Blank,
+            DefinitionStages.WithCreate {
+    }
+
+    /**
+     * Grouping of all the application definition stages.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of the application definition.
+         */
+        interface Blank extends WithSignOnUrl {
+        }
+
+        /**
+         * The stage of application definition allowing specifying the sign on URL.
+         */
+        interface WithSignOnUrl {
+            /**
+             * Specifies the sign on URL.
+             *
+             * @param signOnUrl the URL where users can sign in and use this app
+             * @return the next stage in application definition
+             */
+            WithCreate withSignOnUrl(String signOnUrl);
```

Comment:
so this wither is only available during creation, but there is no corresponding "getter"?  (i.e. something like `String .signOnUrl()`)